sentry_sdk/integrations/gql.py

```diff
@@ -85,25 +90,22 @@ def _patch_execute():
     # type: () -> None
     real_execute = gql.Client.execute
 
+    @ensure_integration_enabled(GQLIntegration, real_execute)
     def sentry_patched_execute(self, document, *args, **kwargs):
         # type: (gql.Client, DocumentNode, Any, Any) -> Any
-        hub = Hub.current
-        if hub.get_integration(GQLIntegration) is None:
-            return real_execute(self, document, *args, **kwargs)
-
-        with Hub.current.configure_scope() as scope:
-            scope.add_event_processor(_make_gql_event_processor(self, document))
+        scope = Scope.get_isolation_scope()
+        scope.add_event_processor(_make_gql_event_processor(self, document))
 
         try:
             return real_execute(self, document, *args, **kwargs)
         except TransportQueryError as e:
             event, hint = event_from_exception(
                 e,
-                client_options=hub.client.options if hub.client is not None else None,
```

Comment:
It looks to me like this `None` check should be unnecessary with the new API – is this correct?

Reply:
Yes, the client has now always options. But please use `sentry_sdk.get_client()` to get the client, because it could be that this scope has no client.